hummingbot/connector/exchange/ndax/ndax_exchange.py

```diff
@@ -310,8 +570,75 @@ def stop_tracking_order(self, order_id: str):
             del self._in_flight_orders[order_id]
 
     async def _update_order_status(self):
+        """
+        Calls REST API to get order status
+        """
         # Waiting on buy and sell functionality.
-        pass
+        tasks = []
+        if len(self._in_flight_orders) < 1:
+            return
+
+        tracked_orders: List[NdaxInFlightOrder] = list(self._in_flight_orders.values())
+        for tracked_order in tracked_orders:
+            ex_order_id = await tracked_order.get_exchange_order_id()
+            if not ex_order_id:
+                self.logger().debug(f"Tracker order {tracked_order.client_order_id} does not have an exchange id. Attempting fetch in next polling interval")
+                continue
+
+            query_params = {
+                "OmsId": 0,
+                "AccountId": self.account_id,
+                "OrderId": ex_order_id,
+            }
+
+            tasks.append(self._api_request(method="GET",
+                                           path_url=CONSTANTS.GET_ORDER_STATUS_PATH_URL,
+                                           params=query_params,
+                                           is_auth_required=True,
+                                           ))
+        self.logger().debug(f"Polling for order status updates of {len(tasks)} orders. ")
+
+        status_responses: List[List[Dict[str, Any]]] = await safe_gather(*tasks, return_exceptions=True)
+
+        # Initial parsing of responses
+        status_responses: List[Dict[str, Any]] = [resp[0]
```

Comment:
I think we should log details if there was an error during the api request (if the task result is an Exception)